### PR TITLE
Port Checkstyle ERT tests to buttercup

### PR DESCRIPTION
**Pull request by Simplify**

### CODE REVIEW

File: test/specs/test-error-parsers.el [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    :buffer 'buffer
+    :line 1
+    :column nil
+    :level 'error
+    :message "Cannot read property 'id' of undefined"
+    :id "Foo1"))
+  "Errors to be parsed from `flycheck-checkstyle-xml'.")
+
+(describe "Error parsers"
+  (describe "Checkstyle XML parsing"
```
Comment (lunaryorn) [inline test/specs/test-error-parsers.el]: Use "The checkstyle parser" as argument for `describe`.  That reads better in the test output imho.

**Comment by lunaryorn** [comment]
@Simplify Great, thanks.

I don't think that we need `defconst` here, though.  As far as I can see there's no reason to have these names available as global variables, so a `let` binding around all `it`s should be enough.

Please use a `let` binding instead of `defconst`s.  I've just refactored the phpmd XML accordingly as well.

**Comment by Simplify** [comment]
@lunaryorn Updated. tslint parser test need updating as well?

**Comment by lunaryorn** [comment]
@Simplify It's buttercup, not butterfly :wink:

**Comment by Simplify** [comment]
@lunaryorn hehehehe, fixed :)

**Comment by lunaryorn** [comment]
@Simplify Thanks.  I've updated the tslint spec.

Please feel free to push simple refactorings in the tests directly to master.  Of course, pull requests are always fine as well, if you'd like to get a review or collect feedback :)

**Comment by Simplify** [comment]
@lunaryorn Yes, I prefer review and feedback until I get more familiar with Flycheck and your vision for this project.

**Pull request closed without merging**